### PR TITLE
Tweel 57 Create Board API

### PR DESCRIPTION
**Pull request by sarahzacky**
Added functionality for the Create Board API so that Workspaces contains an array of boards.

Example of board being added to the database:
image

### CODE REVIEW

File: server/controllers/boardController.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+// @desc    Create a new board
+// @route   POST /api/workspaces/:id/boards
+// @access  Private
+const createBoard = asyncHandler(async (request, response) => {
+  const { name, description } = request.body;
+  const currentUserId = request.user._id;
+  const workspaceId = request.params.workspaceId;
+  
+  console.log(request.params.workspaceId);
```
Comment (danielskjo) [inline server/controllers/boardController.js]: Let's remove all console logs

File: server/controllers/boardController.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  // Check if a board with the same name exists in the workspace
+  const boardExists = workspace.boards.find(board => board.name === name);
+
+  if (boardExists) {
+    response.status(400);
+    throw new Error("Board with this name already exists in the workspace");
+  }
+
+  workspace.boards.push({ name:name, description:description });
```
Comment (danielskjo) [inline server/controllers/boardController.js]: Can we format using prettier?

File: server/routes/workspaceRoutes.js [hunk 3]
```diff
@@ -12,9 +13,16 @@ const router = express.Router();
...
   .get(protect, getWorkspace)
   .put(protect, editWorkspace)
   .delete(protect, deleteWorkspace);
 
+// Board Routes
+router.route("/:workspaceId/boards").get(protect, getBoards).post(protect, createBoard);
+router
+  .put('/:workspaceId/boards/:boardId', protect, editBoard)
+  .delete('/:workspaceId/boards/:boardId', protect, deleteBoard)
+  .get('/:workspaceId/boards/:boardId', protect, getBoard);
```
Comment (danielskjo) [inline server/routes/workspaceRoutes.js]:
```suggestion
router.route('/:workspaceId/boards/:boardId')
  .put(protect, editBoard)
  .delete(protect, deleteBoard)
  .get(protect, getBoard);
```

File: server/controllers/boardController.js (hunk 2 above)
Comment (sarahzacky) [reply server/controllers/boardController.js]: Done!

**Review by danielskjo — APPROVED** [review]
LGTM!

**Pull request merged**